spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -207,9 +207,8 @@ public Statistics estimateStatistics() {
       LOG.debug("using table metadata to estimate table statistics");
       long totalRecords = PropertyUtil.propertyAsLong(table.currentSnapshot().summary(),
           SnapshotSummary.TOTAL_RECORDS_PROP, Long.MAX_VALUE);
-      Schema projectedSchema = expectedSchema != null ? expectedSchema : table.schema();
```

Comment:
`expectedSchema` should not be null.
(Previously, I was replacing calls to `table.schema()` with calls to a method that returns the snapshot schema, but it should not be necessary in `SparkBatchScan`.)